org.eclipse.scout.rt.jackson.test/src/test/java/org/eclipse/scout/rt/jackson/dataobject/JsonDataObjectsSerializationTest.java

```diff
@@ -3798,6 +3801,65 @@ public void testDeserializeTypedPojo() throws Exception {
     assertThrows(InvalidTypeIdException.class, () -> s_lenientDataObjectMapper.readValue("{\"_type\":\"Unknown\"}", TestItemPojo.class));
   }
 
+  @Test
+  public void testSerializeDeserializeNestedRawDoTypes() throws Exception {
+    String inputJson = readResourceAsString("TestNestedRawDo.json");
+    TestNestedRawDo testDo = s_dataObjectMapper.readValue(inputJson, TestNestedRawDo.class);
+
+    TestNestedRawDo expected = BEANS.get(TestNestedRawDo.class)
+        .withDoEntity(BEANS.get(TestTypedUntypedInnerDo.class)
+            .withIId(FixtureStringId.of("qualified-string-id-0"))
+            .withStringId(FixtureStringId.of("unqualified-string-id-0")))
+        .withIDataObject(BEANS.get(TestTypedUntypedInnerDo.class)
```

Comment:
Do we require a test case where the IDataObject is not a DoEntity but a collection instead? We shouldn't do that anyway, usually a DoList, DoCollection... will be defined and not a DoValue, so would be okay for me to leave that one out as a test case.

Reply:
I tried this case, but since DoList is final, there is no option create a typed object which extends IDataObject/DoList.
Added some test cases with raw DoList (and other raw DOs) as example for non-supported cases